segmentation/include/pcl/segmentation/extract_clusters.h

```diff
@@ -41,10 +41,148 @@
 
 #include <pcl/pcl_base.h>
 
+#include <pcl/common/utils.h>
 #include <pcl/search/pcl_search.h>
 
 namespace pcl
 {
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/** \brief Decompose a region of space into clusters based on the Euclidean distance between points
+  * \param it cloud iterator for iterating over the points
+  * \param cloud the point cloud message
+  * \param additional_filter_criteria functor which is used as an additonal criteria that all points being added to the cluster must satisfy
+  * \param tree the spatial locator (e.g., kd-tree) used for nearest neighbors searching
+  * \note the tree has to be created as a spatial locator on \a cloud and \a indices
+  * \param tolerance the spatial cluster tolerance as a measure in L2 Euclidean space
+  * \param clusters the resultant clusters containing point indices (as a vector of PointIndices)
+  * \param min_pts_per_cluster minimum number of points that a cluster may contain (default: 1)
+  * \param max_pts_per_cluster maximum number of points that a cluster may contain (default: max int)
+  * \warning It is assumed that that the tree built on the same indices and cloud as passed here,
+  * if that is not the case then things can go very wrong. For speed reasons, we only check the sizes and not the content
+  * \ingroup segmentation
+  */
+  template <typename PointT, typename FunctorT> void
+   extractEuclideanClusters(
+      ConstCloudIterator<PointT> &it, const PointCloud<PointT> &cloud, FunctorT additional_filter_criteria,
+      const typename search::Search<PointT>::Ptr &tree, float tolerance, std::vector<PointIndices> &clusters,
+      unsigned int min_pts_per_cluster = 1, unsigned int max_pts_per_cluster = std::numeric_limits<int>::max())
+  {
+    if (tree->getInputCloud ()->points.size () != cloud.points.size ())
+    {
+      PCL_ERROR ("[pcl::extractEuclideanClusters] Tree built with a different point cloud size (%lu) than the input cloud (%lu)!\n", tree->getInputCloud ()->points.size (), cloud.points.size ());
+      return;
+    }
+
+    // Check if the tree is sorted -- if it is we don't need to check the first element
+    index_t nn_start_idx = tree->getSortedResults () ? 1 : 0;
```

Comment:
The name `getSortedResults` is worse compared to `areResultsSorted`, but the function returns a bool. Just `index_t nn_start_index = tree->getSortedResults();` should suffice, right?

Reply:
As a reader the current form makes it more obvious what the line does - especially since `getSortedResults` has a confusing name. Direct conversion from `bool` to index type does work in this case, but IME it's the kind of conversation which is hard to understand at first sight when reading the code, while having the ternary makes it more obvious. I believe that the optimizer will generate the same code for both.